### DISCUSSION TITLE
Toggle to repeat

### DISCUSSION THREAD

**Discussion by Legionsys** [sezanzeb/input-remapper#580] [Q&A]
I am hoping someone can assist, 

I am trying to change this 

repeat(10,key(KP_Insert).wait(800))

so that it repeats until I press the key again?

**Comment by Hwa-Chi** [comment]
You can use a variable, something like
if_eq($stop,no,
    set(stop,yes),
    set(stop,no)
).
repeat(10,
	if_eq($stop,no,
            key(KP_Insert).wait(800)
).
set(stop,yes)

if there is no stop variable or it has another value, e.g ."yes",  it will set it to "no" ,
but if you press the key again it will stop. At least that's the idea, I haven't tested the macro, but I use a similar one for sticky keys.

**Reply by Legionsys** [reply]
Thank you, unfortunately it seems that the repeat does not react to dynamic variables, once the repeat loop has been engaged it will continue until completed and then reassess when it is called again.
